src/Stratis.Bitcoin.Features.Consensus/Rules/ProvenHeaderRules/ProvenHeaderSizeRule.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.IO;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.ProvenHeaderRules
+{
+    /// <summary>
+    /// Rule to check if the serialized size of the proven header is less than 1 000 512 bytes.
```

Comment:
thinking about this little more, it might be a better idea to check size of each component
- coinstake size is at max 1000000
- block signature i cant remember, but something like 80 ?
- merkle proof no more than 512

reason being to prevent accepting merkle proof with size 1mb itself